lib/SIL/SILConstants.cpp

```diff
@@ -213,6 +236,15 @@ SymbolicValue SymbolicValue::cloneInto(llvm::BumpPtrAllocator &allocator) const{
                                         memObject->getValue(), allocator);
     return getAddress(newMemObject, accessPath, allocator);
   }
+  case RK_Array:
+  case RK_ArrayAddress: {
+    auto elts = getArrayValue();
+    SmallVector<SymbolicValue, 4> results;
+    results.reserve(elts.size());
+    for (auto elt : elts)
```

Comment:
is there a way to avoid allocation on a per-element basis? like if we know all elts are stored contiguously.

if we have an array of ints, and these ints are expressed "inlined", we could do this. but if the are individual integer literal insts, i guess we couldn't.

Reply:
I'm not sure what you mean, but an array/aggregate of SymbolicValues that are each an inline integer just needs an allocation of the SymbolicValues (which are two words).  The arrays are uniqued too, so this doesn't do a copy.

Reply:
sorry "allocation" is the wrong word, given we are using SmallVector.

I'm wondering that instead of doing "push_back" within a loop, can we do something conceptually like:
```
memcpy(results.data(), elts.data(), elts.size() * num_bytes_per_elt);
```

Reply:
One possible use case is for user to write a large const array in swift code, to be promoted to TF graph. But in that case the main cost probably comes from reading/writing that array, generating graph def, etc, so the cost here may not be a big deal. Agree that we should not prematurely optimize this.